core/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -128,7 +131,8 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
                 if (metaData != null) {
                     ShardPath shardPath = null;
                     try {
-                        shardPath = ShardPath.loadShardPath(logger, nodeEnv, shardId, metaData.getSettings());
+                        IndexSettings indexSettings = new IndexSettings(new Index(metaData.getIndex()), Settings.settingsBuilder().put(settings).put(metaData.getSettings()).build(), Collections.EMPTY_LIST);
```

Comment:
wait, if you have stuff in the node settings you really wanna merge it here no? I think this is a bug we should port? if you configure datapath per node?